pkg/types/keyvalue_store.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+import "context"
+
+type KeyValueStore interface {
```

Comment:
@ettec As I said above, I would change the root interface to just store bytes, and maybe add some validation that the bytes are valid JSON.